lib/src/weightRandom.ts

```diff
@@ -1,20 +1,26 @@
+import { WeightRecord } from '../types'
 import { randomFloat } from './randomFloat'
 import { keys } from './utils'
 
 /**
  * @param specs Object containing elements as properties and their weight as value
  */
-export function weightRandom <T extends Record<string, number>> (specs: T) {
+export function weightRandom <T extends string> (specs: WeightRecord<T>): T {
+  const specsKeys = keys(specs)
+
   let totalWeight = 0
-  for (const prop of keys(specs)) {
+  for (const prop of specsKeys) {
     totalWeight += specs[prop]
   }
 
   const value = randomFloat(1)
 
   let sum = 0
-  for (const prop of keys(specs)) {
+  for (const prop of specsKeys) {
     sum += specs[prop] / totalWeight
     if (value <= sum) return prop
   }
+
+  // Should never happen, purely for type safety
+  return specsKeys[specsKeys.length - 1]
```

Comment:
When something shouldn't happen, throwing an error is usually the better choice.

Reply:
Oh, my bad! I also considered leaving the last element out of the `for`, which perhaps would've been a better option 🤔